### PR TITLE
Issue18 - Increase the max message size from 50mB to 512mB. 

### PR DESCRIPTION
**Pull request by morrowc**
There are a few outsized files in the archive, specifically:

```shell  
  241060964 Jan 20 11:45 updates.20220120.1130.bz2
```

So, increase the max message size to accommodate. There's the clear trend here to: "Heck, just set this to 2GB and stop twiddling up a bit every failure..." - Happy to change the proposed 512 to <something else>.

Also, apparently my git history had changes including:

 - adding a new ftp client to the readChannel function
 - making the evalFile channel buffered

Adding the FTP connection was done to address failures during Walk() with concurrent list/retrieve functions. Looking at tcpdump, it appears that the ftp server/client expect a single serial process to progress. There appears to be a limit to the number of data ports the server will allocate per client command connection, that limit is one. If the client asks for more than one data channel, previous data channels are closed causing interruptions or breakage in the upload processing.

### CODE REVIEW

**Comment by morrowc** [comment]
I think this should be non-controversial...
Oh, though this (I think) is doing the file-size change and half of the integration of channel and ftp process management :(

**Comment by morrowc** [comment]
i'm hitting the upload max file size a bunch apparently..
bgpdata/2021.08/UPDATES/updates.20210817.0930.bz2  grpc: received message larger than max (83499113 vs. 52428800)

there's a bunch of this sort of thing apparently... so I'll fix up the current tree and push a replacement server binary shortly.

**Comment by morrowc** [comment]
to be clear, the above errors will not cause failure(s), until after ~50 per thread have happened.
For the current process running 30 threads I think that's 300 failures tolerated.

File: cmd/archive_upload_mass_upload/mass_upload.go [hunk 1]
```diff
@@ -195,42 +212,69 @@ func (c *client) ftpWalk(dir string) {
...
 		if !ok {
 			glog.Error("Channel closed, exiting readChannel.")
 			break
 		}
 
 		fn := strings.TrimLeft(ef.name, "/")
+		// If the file doesn't have UPDATES/updates do not process further.
+		if !strings.Contains(fn, "UPDATES/updates") {
+			continue
+		}
```
Comment (yuhao-he) [inline cmd/archive_upload_mass_upload/mass_upload.go:258]: I suppose we may need this tool for RIPE as well...? I don't think RIPE paths have "UPDATES/updates".

File: cmd/archive_upload_mass_upload/mass_upload.go [hunk 2]
```diff
@@ -295,14 +343,22 @@ func main() {
...
 
+	// Start the readChannel threads.
+	for i := 0; i < *threads; i++ {
+		go c.readChannel(ctx)
+	}
+
 	// Start the FTP walk, then read from the channel and evaluate each file.
 	go c.ftpWalk(dir)
-	c.readChannel(ctx)
+	// c.readChannel(ctx)
```
Comment (yuhao-he) [inline cmd/archive_upload_mass_upload/mass_upload.go:358]: nit: should we just delete this?

**Review by yuhao-he — APPROVED** [review]

File: cmd/archive_upload_mass_upload/mass_upload.go (hunk 1 above)
Comment (morrowc) [reply cmd/archive_upload_mass_upload/mass_upload.go:258]: Good point.. I imagine a map of:
  proto.DATASOURCE -> important-path-attribute

is in our future :) Ripe appears to store in this central location:
  https://data.ris.ripe.net/rrc11/2021.03/updates.20210331.225.gz
  
  rrc11 == 1 collector name... there are ~30-odd collectors.
Before we start digesting that data we should chat up their lead for these things, who I think is:
   Robert Kisteleki (he's easy to get ahold of)

File: cmd/archive_upload_mass_upload/mass_upload.go (hunk 2 above)
Comment (morrowc) [reply cmd/archive_upload_mass_upload/mass_upload.go:358]: done.

**Pull request merged**